src/relay/transforms/annotate_target.cc

```diff
@@ -93,18 +106,20 @@ class AnnotateTargetRewriter : public ExprRewriter {
       if (ref_target == "") {
         ref_target = arg_target;
       } else if (ref_target != arg_target) {
-        ref_target = "default";
+        ref_target = default_target;
       }
     }
 
     // Determine compiler begin target.
     std::string op_target = (target == "") ? ref_target : target;
 
-    Array<Expr> compiler_begins;
-    for (const auto& end : compiler_ends) {
-      compiler_begins.push_back(InsertAnnotation(end, op_target, make_begin_op));
+    if (ref_target != "") {
+      for (const auto& end : compiler_ends) {
+        compiler_begins.push_back(InsertAnnotation(end, op_target, make_begin_op));
+      }
+    } else {
+      return {op_target, args};
```

Comment:
This makes L188 unreachable?

Reply:
I do not understand you comment. please elaborate.

Reply:
nvm. I figured it out.